src/Discord.Net/Entities/Webhooks/IWebhook.cs

```diff
@@ -0,0 +1,25 @@
+// todo: docs
+using System.Threading.Tasks;
+
+namespace Discord
+{
+    public interface IWebhook
+    {
+        ValueTask<IUser> GetCreatorAsync(StateBehavior stateBehavior = default, RequestOptions? options = default);
+        ValueTask<ITextChannel> GetChannelAsync(StateBehavior stateBehavior = default, RequestOptions? options = default);
+        ValueTask<IGuild> GetGuildAsync(StateBehavior stateBehavior = default, RequestOptions? options = default);
+        // review: does this need to be an async retrieval? the only path to get an IWebhook should be via an ITextChannel
+
+        string Token { get; }
+
+        // note: yeah, the full creator is included in the payload, but we don't want entities to hold direct references to other entities
```

Comment:
I can get behind that. So long as there's a hot path optimization within `GetCreatorAsync()` if there's potential for an `async` call when fresh data has just come in on the same message.